### PR TITLE
Updates to support esp-idf v5.0 on CircuitPython

### PR DESCRIPTION
**Pull request by MicroDev1**

### CODE REVIEW

File: src/arch/esp32-s3.h [hunk 1]
```diff
@@ -279,12 +283,6 @@ void _PM_timerInit(Protomatter_core *core) {
   desc.buffer = core->screenData;
   desc.next = NULL;
 
-  // Alloc DMA channel & connect it to LCD periph
-  gdma_channel_alloc_config_t dma_chan_config = {
-      .sibling_chan = NULL,
-      .direction = GDMA_CHANNEL_DIRECTION_TX,
-      .flags = {.reserve_sibling = 0}};
-  esp_err_t ret = gdma_new_channel(&dma_chan_config, &dma_chan);
```
Comment (dhalbert) [inline src/arch/esp32-s3.h:287]: Not sure why this was removed?

File: src/arch/esp32.h [hunk 2]
```diff
@@ -63,18 +65,6 @@
...
-// actually atomic. If two writes are made in quick succession, the second
-// has no effect. One option is NOPs, other is to write a 0 (no effect) to
-// the opposing register (set vs clear) to synchronize the next write.
-#define PEW                                                                    \
-  *set = (*data++) << shift; /* Set RGB data high */                           \
-  *clear_full = 0;           /* ESP32 MUST sync before 2nd 'set' */            \
-  *set = clock;              /* Set clock high */                              \
-  *clear_full = rgbclock;    /* Clear RGB data + clock */                      \
-  ///< Bitbang one set of RGB data bits to matrix
-
```
Comment (dhalbert) [inline src/arch/esp32.h:77]: Also why this was removed?

**Review by dhalbert — CHANGES_REQUESTED** [review]

File: src/arch/esp32-s3.h (hunk 1 above)
Comment (MicroDev1) [reply src/arch/esp32-s3.h:287]: These variables are initialized but left unutilized.

File: src/arch/esp32.h (hunk 2 above)
Comment (dhalbert) [reply src/arch/esp32.h:77]: PEW is used in `src/core.c` and mentioned in comments elsewhere.

**Review by dhalbert — CHANGES_REQUESTED** [review]

File: src/arch/esp32-s3.h (hunk 1 above)
Comment (MicroDev1) [reply src/arch/esp32-s3.h:287]: Good Catch! I only meant to remove the unutilized `esp_err_t ret` part.

File: src/arch/esp32.h (hunk 2 above)
Comment (MicroDev1) [reply src/arch/esp32.h]: Fixed now.

**Review by dhalbert — APPROVED** [review]
OK, these changes look fine to me. @MicroDev1 do you have anything you can test this with (an RGB Matrix, etc.)? They could be tested in ESP Arduino without CircuitPython.

**Comment by MicroDev1** [comment]
I believe, I accidentally removed @PaintYourDragon from review.
 
> do you have anything you can test this with (an RGB Matrix, etc.)?

I can't test as I don't have the hardware.

**Review by PaintYourDragon — APPROVED** [review]
Tested Arduino lib on Feather ESP32-S2, seems OK, haven’t tested S3 but since this seems to be mostly timer related I’m pretty sure it’ll work all the same.

**Comment by dhalbert** [comment]
Thanks for the quick test @PaintYourDragon!

**Comment by PaintYourDragon** [comment]
Oh also — is matrix support for both S2 _and S3_ canonically in CircuitPython at this point? I have a to-do elsewhere that could be closed out in that case. Thx!

**Comment by dhalbert** [comment]
> Oh also — is matrix support for both S2 _and S3_ canonically in CircuitPython at this point? I have a to-do elsewhere that could be closed out in that case. Thx!

It is turned off for plain ESP32 but otherwise is on for everything else: https://github.com/adafruit/circuitpython/blob/b6f67be3e2c58d3780e2a1bc3801ac5a43f19d4d/ports/espressif/mpconfigport.mk#L40

I'm not actually sure why it's turned off for plain ESP32. Is that your doing? Plain ESP32 has various IDF differences.

**Comment by PaintYourDragon** [comment]
Not my doing, I’ve tended to stick to the Arduino side of the code. I could speculate that it might be too much for the OG ESP32 (since the code there is all brute force bitbang-y, no peripherals/DMA), but not sure.

OR…possibly it was meant to be disabled for the C3 and something went in the wrong place. I don’t think there’s enough GPIO on the C3 to run a matrix, and even if there was, looks like the code’s empty for that one, like it might’ve just been a placeholder or something put there to pass CI.

**Comment by PaintYourDragon** [comment]
Any relationship between this and PR https://github.com/adafruit/Adafruit_Protomatter/pull/47? Looks like changes from the latter have already been incorporated and it can be closed? Not sure.

**Comment by jepler** [comment]
You can close up #47, because I'm not likely to get back to testing on that microcontroller anytime soon. When I last tested on circuitpython it didn't work or else my testing setup was wrong

**Pull request merged**